speechless/processing/analysis/tfidf.py

```diff
@@ -0,0 +1,100 @@
+from argparse import ArgumentParser
+from logging import Logger
+from pathlib import Path
+from typing import List
+
+import gensim.downloader
+from gensim.corpora import Dictionary
+from gensim.models import TfidfModel
+
+from speechless.edit_context.common import TimelineChange
+from speechless.processing.analysis.analysis import (ARG_PREPARE_METHOD, AnalysisDomain,
+                                                     AnalysisMethod, analysis_method_cli)
+from speechless.processing.tokenization import EditToken, sentence_segmentation
+from speechless.readers.subtitles import vtt_reader
+from speechless.utils.logging import NULL_LOGGER
+
+DICTIONARY_FILE = 'tfidf_dictionary.dic'
+MODEL_FILE = 'tfidf_model.model'
+
+
+class TfidfAnalysis(AnalysisMethod):
+
+  def __init__(self, dataset: str, sentence_threshold: float, logger: Logger = NULL_LOGGER) -> None:
+    super().__init__('Tf-idf Analysis', [AnalysisDomain.TEXT])
+    self.logger = logger
+    self.sentence_threshold = sentence_threshold
+    gensim.downloader.BASE_DIR = '/opt/workspace/SpeechLess/gensim-data'
+    gensim.downloader.base_dir = '/opt/workspace/SpeechLess/gensim-data'
```

Comment:
These paths didn't work on my laptop

Reply:
In MxNet, we create a directory at the user's home directory for such files (~/.mxnet/, so ~/.speechless/ in our case)